app/src/main/java/com/github/libretube/api/PlaylistsHelper.kt

```diff
@@ -175,7 +175,13 @@ object PlaylistsHelper {
             awaitQuery {
                 DatabaseHolder.Database.localPlaylistsDao().removePlaylistVideo(transaction.videos[index])
             }
-            if (transaction.videos.size > 1) return
+            if (transaction.videos.size > 1) {
+                if (index == 0) {
+                    transaction.videos[1].thumbnailUrl?.let { transaction.playlist.thumbnailUrl = it }
+                    DatabaseHolder.Database.localPlaylistsDao().updatePlaylist(transaction.playlist)
```

Comment:
That seems to be running on the main thread, wouldn't that crash?

Reply:
Yeh, should be awaitQuery.